pkg/destinations/redshift/insert.go

```diff
@@ -0,0 +1,149 @@
+package redshift
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/scratchdata/scratchdata/util"
+
+	"github.com/scratchdata/scratchdata/pkg/storage/blobstore/s3"
+)
+
+func (s *RedshiftServer) createColumns(table string, jsonTypes map[string]string) error {
+
+	for colName, jsonType := range jsonTypes {
+		colType := "VARCHAR"
+		switch jsonType {
+		case "int":
+			colType = "BIGINT"
+		case "bool":
+			colType = "BOOLEAN"
+		case "float":
+			colType = "DOUBLE PRECISION"
+		case "string":
+			colType = "VARCHAR"
+		default:
+			colType = "VARCHAR"
+		}
+
+		sql := fmt.Sprintf("ALTER TABLE \"%s\" ADD COLUMN \"%s\" %s", table, colName, colType)
+		_, err := s.conn.Exec(sql)
+		if err != nil {
+			if !strings.Contains(err.Error(), "already exists") {
+				log.Err(err).Msg("createColumns: cannot create column")
+				return err
+			}
+
+		}
+		log.Info().Msgf("Created column %s %s", colName, colType)
+		// if _, ok := cols[strings.ToLower(colName)]; !ok {
+
+		// }
+
+	}
+
+	return nil
+}
+func (s *RedshiftServer) CreateColumns(table string, fileName string) error {
+
+	input, err := os.Open(fileName)
+	if err != nil {
+		return err
+	}
+	// Infer JSON types for the input
+	jsonTypes, err := util.GetJSONTypes(input)
+	if err != nil {
+		return err
+	}
+
+	err = s.createColumns(table, jsonTypes)
+	if err != nil {
+		return err
+	}
+
+	err = input.Close()
+	if err != nil {
+		log.Error().Err(err).Str("filename", fileName).Msg("Unable to close file")
+	}
+
+	return nil
+
+}
+
+func (s *RedshiftServer) CreateEmptyTable(table string) error {
+
+	sql := fmt.Sprintf("CREATE TABLE IF NOT EXISTS \"%s\" (__row_id BIGINT)", table)
+	_, err := s.conn.Exec(sql)
+	return err
+}
+
+func (s *RedshiftServer) InsertFromNDJsonFile(table string, filePath string) error {
+	// Make sure the table exists
+
+	// Recalling createColumns to create columns in the table if missing,  will be created
+	err := s.CreateColumns(table, filePath)
+	if err != nil {
+		return err
+
+	}
+
+	params := make(map[string]any)
+
+	params["region"] = s.S3Region
+	params["access_key_id"] = s.S3AccessKeyId
+	params["secret_access_key"] = s.S3SecretAccessKey
+	params["bucket"] = s.S3Bucket
+	params["skipDefaultConfig"] = true
```

Comment:
I don't understand the reason for `skipDefaultConfig`. Can you describe why we need this check? Could we simplify the code if we don't use it?